### PR TITLE
FDT fix

### PR DESCRIPTION
**Pull request by hlyytine**
Test-with: tiiuae/camkes-vm#23
Test-with: tiiuae/sel4_projects_libs#20

### CODE REVIEW

File: include/ioreq.h [hunk 1]
```diff
@@ -35,8 +37,7 @@ typedef struct io_proxy {
     void (*backend_notify)(struct io_proxy *io_proxy);
     int (*run)(struct io_proxy *io_proxy);
     uint32_t backend_id;
-    uintptr_t data_base;
-    size_t data_size;
+    fdt_reserved_memory_t *data_plane;
```
Comment (elmankku) [inline include/ioreq.h:40]: This effectively blocks using non-generated DTBs and configuration without SWIOTLB, and I think the biggest problem is that it couples configuration and operation.

I don't see a reason why this should be part of the io_proxy struct. We could use another struct which is f.ex. registered (or simply an array that is generated) and matched using `backend_id` when a node needs to be generated.

**Review by elmankku — COMMENTED** [review]
I think we should try to keep configuration and normal operation decoupled. Check the comment.

File: include/ioreq.h (hunk 1 above)
Comment (hlyytine) [reply include/ioreq.h]: How about now?

**Review by elmankku — APPROVED** [review]
Good :+1:

**Pull request merged**